File: src/joint.cpp
#include "joint.h"
//	#include	"ODrive.h"


/*
	@param: odrive_serial, odrive
*/
	// NOTE: Should constructor has a motor_number param
Joint::Joint() {
	/*
		NOTE: How to check wether calibration needs to happen?
	*/
	odrive_serial = new SoftwareSerial(10, 11);
	odrive = new ODriveArduino(*odrive_serial);

	Serial.begin(9600);
	while (!Serial) ; // wait for Arduino Serial Monitor to open

	odrive_serial->begin(115200); 	// ODrive uses 115200 baud
	// while (!odrive_serial);
	Serial.println(HEADER);
}
/*
  	@brief: runs initial calibration of the DC motor and odrive
*/
void Joint::calibrate(int motor_number) {
	odrive->Calibrate(motor_number);
}

/*
	 TODO: Write update, set_position, and get_position methods
*/
void Joint::update() {
	// use set_position and get_position for pd control
}

/*
	@param: double pos
*/
void Joint::set_position() {

}

/*
	@return: double pos
*/
void Joint::get_position() {

}

File: legacy/python/src/joint.py
from collections import namedtuple

gains = namedtuple('gains', 'kp ki kd')  # namedtuple is immutable


class Joint:
    '''
        @param: ODrive oobject, list gains, boolean calibrate
    '''

    def __init__(self):
        '''
            class attributes:
            ODrive = oobject;
            g.kp = gains[0];
            g.ki = gains[1];
            g.kd = gains[2];
            target_position;
            current_position;

            check whether calibration needs to happen during initialization
        '''
        pass

    '''
        @brief: runs initial calibration of the brushless DC motor and position encoder 
    '''

    def calibrate(self):
        pass

    '''
        @brief: use set_position and get_position for pd control
    '''

    def update(self):
        pass

    '''
        @param: float pos
    '''

    def set_position(self):
        pass

    '''
        @return: float pos
    '''

    def get_position(self):
        pass
